h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -279,6 +424,17 @@ public synchronized TwoDimTable to2dTable() {
         res.set(i, col++, _scoringIters.get(i));
         res.set(i, col++, _likelihoods.get(i));
         res.set(i, col++, _objectives.get(i));
+        if (parms._generate_scoring_history) {
+          res.set(i, col++, _alphas.get(i));
+          res.set(i, col++, _lambdas.get(i));
+          res.set(i, col++, _lambdaDevTrain.get(i));
+          if (_lambdaDevTest != null) 
+            res.set(i, col++, _lambdaDevTest.get(i));
```

Comment:
with `validation_frame` specified this works

Reply:
Yes, I fixed this error.  The array deviance test is instantiated but not added to when there is no validation dataset.   Hence, there is an NPE error when it was accessed (since deviance_test != null will be true).  When you have a validation dataset, since the test deviance is instantiated and is added to, hence there should be no error when you try access it.